packages/gatsby/cache-dir/head/head-export-handler-for-browser.js

```diff
@@ -84,6 +86,40 @@ if (process.env.BUILD_STAGE === `develop`) {
   })
 }
 
+// TODO(serhalp): Expose these as a new public API to allows us to gradually phase out
+// the existing `<html>` and `<body>` features in the Head API, allowing us to remove
+// the various workounds we have in place for React 19.
+function Html(props) {
+  return <div data-original-tag="html" {...props} />
+}
+function Body(props) {
+  return <div data-original-tag="body" {...props} />
+}
+
+// React 19 does not allow rendering a second `<html>` or `<body>` anywhere in the tree:
+// https://github.com/facebook/react/blob/50e7ec8a694072fd6fcd52182df8a75211bf084d/packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js#L3739
+// Unfortunately, our Head API relies on users being able to render these elements, which we then
+// render in a hidden `<div>` to extract attributes to apply to the real `<html>` and `<body>`.
+// We explored several alternatives, but none were viable given React's constraints.
+// To work around this, we intercept attempts to render these elements inside of
+// Head and replace them with custom components that render as `<div>`s with a
+// `data-original-tag` attribute. We can then use this attribute later to apply
+// attributes to the real `<html>` and `<body>` elements.
+const IsHeadRenderContext = React.createContext(false)
+// De-risk monkey patch by only applying it when needed
+if (semver.gte(React.version, `19.0.0`)) {
```

Comment:
This probably is adding `semver` to browser bundle (unless it was there already). 

If we need to do semver check, it's better to do it at some point in Node land and use `webpack.DefinePlugin` to be able to do simple truthy checks in browser runtime

Reply:
I checked and as-is this is what we get in browser bundles (for hello world without additional dependencies other than gatsby and react@18 itself) without the changes in this PR:

<img width="2560" height="1328" alt="Image" src="https://github.com/user-attachments/assets/c5f6b04b-e849-4fd8-93b9-c0cd6adebf91" />

And this is with changes in PR:

<img width="2560" height="1328" alt="Image" src="https://github.com/user-attachments/assets/8f5f9866-0ff3-40d4-bbb5-10739ad7d6e0" />

`semver` adds ~7kb gzipped size to bundles

Reply:
I suspect simplest solution would be to still check `React.version`, but avoid using `semver` package and just pick up major in a bit more manual way? With either splitting version string in `.` and just picking first value or some regex test?